pootle/apps/virtualfolder/utils.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+from fnmatch import fnmatch
+
+from pootle.core.url_helpers import split_pootle_path
+from pootle_store.models import Store
+
+
+class VirtualFolderPathMatcher(object):
+
+    def __init__(self, vf):
+        self.vf = vf
+
+    def update_matching_stores(self):
+        existing_stores = self.vf.stores.all()
+        matching_stores = self.find_matching_stores()
+        to_add = []
+        to_delete = []
+        for store in matching_stores:
+            if store not in existing_stores:
+                to_add.append(store)
+        for store in existing_stores:
+            if store not in matching_stores:
+                to_delete.append(store.id)
+        if to_add:
+            self.vf.stores.add(*to_add)
+        if to_delete:
+            self.vf.stores.remove(
+                *self.vf.stores.filter(id__in=to_delete))
+        return to_add, to_delete
+
+    def add_store_if_matching(self, store):
+        pootle_dir_path = "".join(
+            ("/", ) + split_pootle_path(store.pootle_path)[2:])
+        no_add = (
+            not self.path_matches(pootle_dir_path)
```

Comment:
If all rules are directory paths, and this store is within one of those directories, then the store won't be added, am I wrong?